diff -ur stock/grep-2.5.4/po/Makefile.in.in grep-2.5.4/po/Makefile.in.in
--- stock/grep-2.5.4/po/Makefile.in.in	Tue Feb  3 10:49:43 2009
+++ grep-2.5.4/po/Makefile.in.in	Wed Jan 18 17:57:49 2023
@@ -14,6 +14,7 @@
 
 srcdir = @srcdir@
 top_srcdir = @top_srcdir@
+top_builddir = @top_builddir@
 VPATH = @srcdir@
 
 prefix = @prefix@
diff -ur stock/grep-2.5.4/src/grep.c grep-2.5.4/src/grep.c
--- stock/grep-2.5.4/src/grep.c	Tue Feb  3 10:49:52 2009
+++ grep-2.5.4/src/grep.c	Wed Jan 18 17:39:13 2023
@@ -1860,7 +1860,7 @@
   textdomain (PACKAGE);
 #endif
 
-  atexit (close_stdout);
+  /* atexit (close_stdout); */
 
   prepend_default_options (getenv ("GREP_OPTIONS"), &argc, &argv);
 
