diff --exclude=Makefile* --exclude=c_rehash --exclude=opensslconf.h --exclude=CA.pl -ur stock/openssl-1.0.2o/crypto/ui/ui_openssl.c openssl-1.0.2o/crypto/ui/ui_openssl.c
--- stock/openssl-1.0.2o/crypto/ui/ui_openssl.c	Tue Mar 27 05:54:46 2018
+++ openssl-1.0.2o/crypto/ui/ui_openssl.c	Wed Jan 18 23:37:44 2023
@@ -266,6 +266,10 @@
 typedef int sig_atomic_t;
 #endif
 
+#if defined(_AIX) && defined(__i386__)
+typedef int sig_atomic_t;
+#endif
+
 #if defined(OPENSSL_SYS_MACINTOSH_CLASSIC) || defined(MAC_OS_GUSI_SOURCE) || defined(OPENSSL_SYS_NETWARE)
 /*
  * This one needs work. As a matter of fact the code is unoperational
diff --exclude=Makefile* --exclude=c_rehash --exclude=opensslconf.h --exclude=CA.pl -ur stock/openssl-1.0.2o/e_os.h openssl-1.0.2o/e_os.h
--- stock/openssl-1.0.2o/e_os.h	Tue Mar 27 05:54:46 2018
+++ openssl-1.0.2o/e_os.h	Wed Jan 18 23:49:23 2023
@@ -775,6 +775,14 @@
 #  endif
 # endif
 
+# if defined(_AIX) && defined(__i386__)
+#  include <string.h>
+#  include <memory.h>
+void bcopy (char *source, char *target, int n);
+#  define memmove(s1,s2,n) bcopy((s2),(s1),(n))
+#  define strtoul(s,e,b) ((unsigned long int)strtol((s),(e),(b)))
+# endif
+
 #ifdef  __cplusplus
 }
 #endif
diff --exclude=Makefile* --exclude=c_rehash --exclude=opensslconf.h --exclude=CA.pl -ur stock/openssl-1.0.2o/e_os2.h openssl-1.0.2o/e_os2.h
--- stock/openssl-1.0.2o/e_os2.h	Tue Mar 27 05:54:46 2018
+++ openssl-1.0.2o/e_os2.h	Wed Jan 18 23:37:44 2023
@@ -318,6 +318,10 @@
 #  define ossl_ssize_t int
 # endif
 
+# if defined(_AIX) && defined(__i386__) && !defined(ssize_t)
+#  define ossl_ssize_t long int
+# endif
+
 # ifndef ossl_ssize_t
 #  define ossl_ssize_t ssize_t
 # endif
diff --exclude=Makefile* --exclude=c_rehash --exclude=opensslconf.h --exclude=CA.pl -ur stock/openssl-1.0.2o/engines/ccgost/gosthash.c openssl-1.0.2o/engines/ccgost/gosthash.c
--- stock/openssl-1.0.2o/engines/ccgost/gosthash.c	Tue Mar 27 05:54:46 2018
+++ openssl-1.0.2o/engines/ccgost/gosthash.c	Thu Jan 19 00:04:30 2023
@@ -11,6 +11,8 @@
 #include "gost89.h"
 #include "gosthash.h"
 
+#include "../e_os.h"
+
 /*
  * Use OPENSSL_malloc for memory allocation if compiled with
  * -DOPENSSL_BUILD, and libc malloc otherwise
